runtime.go

```diff
@@ -240,19 +239,22 @@ func (r *Runtime) Invoke(event string) (io.Reader, io.Reader, error) {
 	log.Printf("Invoking %s (%s)\n", r.Function.Handler, r.Name)
 
 	// If the CodeUri has been specified as a .jar or .zip file, unzip it on the fly
-	if strings.HasSuffix(r.Cwd, ".jar") || strings.HasSuffix(r.Cwd, ".zip") {
-		log.Printf("Decompressing %s into runtime container...\n", filepath.Base(r.Cwd))
-		decompressedDir, err := decompressArchive(r.Cwd)
-		if err != nil {
-			log.Printf("ERROR: Failed to decompress archive: %s\n", err)
-			return nil, nil, fmt.Errorf("failed to decompress archive: %s", err)
+	if r.Function.CodeUri != nil && r.Function.CodeUri.String != nil {
+		codeuri := *r.Function.CodeUri.String
```

Comment:
It will be good to move this into the Function itself. Let the function return a path where code is found. It can also have a cleanup() method that `invoke` will call to cleanup the temp dir